appserver/data/data-core/src/main/java/fish/payara/data/core/util/DataCommonOperationUtility.java

```diff
@@ -421,4 +427,27 @@ public static DataParameter extractDataParameter(Object[] args) {
         }
         return new DataParameter(limit, sortList);
     }
+
+    public static void startTransactionAndJoin(TransactionManager transactionManager,
+                                               EntityManager em, boolean isUserManagedTransaction) throws SystemException, NotSupportedException {
+        
+        if (isUserManagedTransaction) {
+            return;
+        }
+        
+        if (transactionManager.getStatus() == jakarta.transaction.Status.STATUS_NO_TRANSACTION) {
+            transactionManager.begin();
+            em.joinTransaction();
```

Comment:
You are joining twice? Why not delete this line?

Reply:
already done!!